packages/ra-core/src/form/Form.stories.tsx

```diff
@@ -164,3 +166,25 @@ export const UndefinedValue = () => {
         </CoreAdminContext>
     );
 };
+
+const zodSchema = z.object({
+    preTranslated: z.string().min(5, { message: 'Required' }),
+    translationKey: z.string().min(5, { message: 'ra.validation.required' }),
+});
+export const ZodResolver = () => {
+    const [result, setResult] = React.useState<any>();
+    return (
+        <CoreAdminContext>
+            <Form
+                record={{}}
+                onSubmit={data => setResult(data)}
+                resolver={zodResolver(zodSchema)}
+            >
+                <Input source="preTranslated" />
+                <Input source="translationKey" />
```

Comment:
it's not entirely clear what to expect here. Would you mind adding a description?